tedge/src/cli/connect/command.rs

```diff
@@ -165,6 +174,22 @@ impl ConnectCommand {
             Cloud::C8y => check_device_status_c8y(port),
         }
     }
+
+    fn check_if_bridge_exists(&self, br_config: BridgeConfig) -> bool {
+        // Check if bridge exists and stop with code 0 if it doesn't.
+        let bridge_conf_path = self
+            .config_location
+            .tedge_config_root_path
+            .join(TEDGE_BRIDGE_CONF_DIR_PATH)
+            .join(br_config.config_file);
+
+        match Path::new(&bridge_conf_path).exists() {
+            // If we find the bridge config file we remove it
+            // and carry on to see if we need to restart mosquitto.
+            true => true,
+            false => false,
+        }
+    }
```

Comment:
I have a negative feeling about this kind of check. That code is supposed to have created that file. If not it should have returned an error *before* running that test that is useless.

Furthermore:
* The comment is misleading: do we really remove the file it exists?
* A match true => true, false => false, is useless.

Reply:
I removed those comments, and fixed the code logic, and simplified the code. The whole purpose here is to check locally if the device is connected to the cloud or not before doing the actual check_connection. With the existing code even if the device is not connected still goes ahead and publishes the messages on the topic, if we do this check before we can avoid this.